+	4 Sequence dataset constructionAll known complete or nearly complete genome sequences of Hepatitis E virus (Paslahepevirus balayani) were extracted from the <database>GenBank</database> and <database>ViPR</database> databases (identified using a combination of <database>BLAST</database> and taxonomy searching) and placed into a dataset containing our genomes from the 2023 South Sudan outbreak.
+	Annotations of protein families were generated using the Virus Pathogen Resource (<database>ViPR</database>) at www.
+	We downloaded all full ZIKV genomes from the <database>ViPR</database> database [40] deposited until May 2017.
+	, 2004), <database>EuPathDB</database> (4formerly <database>ApiDB</database>), <database>WormBase</database>5, Virus Pathogen Database and Analysis Resource (<database>ViPR</database>) has led to tremendous advances in the search for new drug and vaccine targets (Yan et al.
+	A second source was the HCV genotyping procedure of the Virus Pathogen Database and Analysis Resource(<database>ViPR</database> [8]).
+	We also obtained genotype and subtype results generated by the <database>ViPR</database> HCV genotyping method for each sequence.
+   Both the <database>ViPR</database> HCV resource [7] and the <database>REGA</database> tools [1] provide a genotyping service for submitted sequences.
+	The reference sequence (Sequence name: 10MYKJ037) was downloaded from Virus Pathogen Database and Analysis Resource(<database>ViPR</database>) [34], which is a partial coding DNA sequence (CDS) of HPgV isolated from Malaysia in 2010.
+	In Africa, apart from one metapneumovirus genome identified from a wild mountain gorilla in Rwanda (<database>GenBank</database> accession number HM197719), there are no HMPV genomes reported according to the NIAID Virus Pathogen Database and Analysis Resource(<database>ViPR</database>, http://www.
+	, identical sites averaging over all sequence pairs and excluding positions containing gaps) between all 143 genome sequences analyzed (5 new genomes plus 138 from <database>ViPR</database>) was 58.
+	Phylogenetic analyses A dataset of HMPV genome sequences was retrieved from <database>ViPR</database> in order to infer relationship between HMPV viruses from Kenya and Zambia and viral populations sampled globally.
+	<database>nextstrain</database>,<database>covidex</database>,<database>GESS</database>,<database>pangolin</database>, COVID-19 genotyping tool, SARS-CoV-2 alignment screen, <database>UCSC</database> SARS-CoV-2 genome browser, 2019nCoVR, <database>CoV2ID</database>, COVID-19 data portal, </database>NCBI Database</database> nucleotide sequences, SARS-CoV-2 data hub, <database>ViPR</database> SARS-CoV-2, WOLFRAM Protein	Protein sequence, protein structure, PDB protein structures, protein expression, protein–protein interactions (PPI), function, localization, protein mutations, protein alignment, protein domains.
+	<database>CoViProteins</database>, <database>coronavirus3D</database>, <database>neXtProt</database>, COVID-19 docking server, COVID-19 molecular structure and therapeutics hub, COVID-19 <database>UniProtKB</database>, <database>COVIDep</database>, </database>NCBI Database</database> protein sequences, <database>PROSITE</database>, <database>PubChem</database> COVID-19 data, <database>STRING</database> COVID-19 host-interactome, <database>SWISS-MODEL</database>, the human protein atlas, <database>VBRC</database>, <database>ViPR</database> NCBI-SARS-CoV-2, <database>Virus-CKB</database> <database>RNA</database>	RNA-seq, RNA secondary structures, conservation, RNA motifs, RNA structures.
+	MethodsProteomic data retrieval<database>ViPR</database> database was used as it is based on <database>IEDB</database> for database assessment that assisted us in gathering information regarding viral proteome.
+	<database>ViPR</database> database was used for gathering information regarding viral proteome.
+	, the Virus Pathogen Database and Analysis Resource, <database>ViPR</database> (Pickett et al.
+	Protein Identification and RetrievalThe following 12 viral protein sequences of SARS-CoV-2 were retrieved from the <database>ViPR</database> database (Host: Human, Country: India) [10]: Orf10 protein (QIA98591.
+	ResultsSelection and Retrieval of Viral Protein SequencesThe SARS-CoV-2 strain was identified and 12 viral protein sequences against the human host in India were retrieved from the <database>ViPR</database> database and selected for possible vaccine candidate identification (Table 1).
+	Table 1 SARS-CoV-2 (Host: Human, Country: India) viral protein sequence identification and retrieval via the <database>ViPR</database> database.
+	All data sets and associated metadata have been submitted to Virus Pathogen Resource(<database>ViPR</database>, http://www.
+	FASTA files were generated using the virus pathogen database and analysis resource (<database>ViPR</database>) [16].
+	The integrated SFTSV database was constructed by combining the protein sequences which were downloaded from the <database>ViPR</database> (https://www.
+	Materials and methodsGenetic variability analysisA total of 5885 nucleotide sequences-data set (2086 of DENV-1, 1625 of DENV-2, 1516 of DENV-3 and 658 of DENV-4) including four DENV reference sequences, from the Virus Pathogen Database and Analysis Resource(<database>ViPR</database>) of the National Institute of Allergy and Infectious Diseases (<database>NIH</database>/<database>DHHS</database>) (accessed March 2020), were initially included in an analysis of genetic variability.
+	To construct these data sets, the genotype of each of the 5885 <database>ViPR</database> sequences was determined, grouped according to genotype, and redundant sequences were removed to avoid overrepresentation of sequences due to sampling.
+	, 2022), Virus Pathogen Resource(<database>ViPR</database>) (Pickett et al.
+	, 2022), <database>pyViPR</database> (Ortega and Lopez, 2020) and <database>DIVIPAC</database> (Nguyen et al.
+	The Pathosystems Resource Integration Center (<database>PATRIC</database>) and the Virus Pathogen Database and Analysis Resource(<database>ViPR</database>) also provide standardized metadata for some pathogenic bacterial and viral genomes, respectively [6, 7].
+	In this respect is it similar to capabilities offered in <database>ViPR</database> for some virus genomes [6], and in <database>PATRIC</database> for assembled bacterial genomes [7], but provides access to all sample entries in <database>BioSample</database> including for the growing number associated with unassembled genomic data.
+	According to the Virus Pathogen Database and Analysis Resource(<database>ViPR</database>)([15]), Rhabdoviridae is divided into 12 genera and for ease of exposition we have limited our analysis to the data set containing all distinct and complete genomes from those genera with less than 40 complete genomes (9 of 12 genera).
+	py written in Python from <database>ViPR</database> software (https://github.
+	To further investigate these sequences, segment-specific databases were compiled for the potential reassortants, Vietnamese G1P8 and G2P4 sequences with a standard Wa-like or DS-1-like genome constellation, and a representative subsample of G1P8 RoV with a Wa-like backbone (I1-R1-C1-M1-A1-N1-T1-E1-H1) and G2P4 RoV with a DS-1-like backbone were obtained from the Virus Pathogen Resource(<database>ViPR</database>) database (http://www.
+	gz, which contains 2042 Dengue Type 1 genomes downloaded from the Virus Pathogen Database and Analysis Resource(<database>ViPR</database>) (Pickett et al.
+	cn/ncov) [80], NIAID Virus Pathogen Resource(<database>ViPR</database>) database (https://www.
+	Supplementary data 11 Acknowledgments The authors would like to thank to all authors, originating and submitting laboratories that share their sequences through the  <database>EpiCoV</database>™, <database>GenBank</database>, 2019nCoVR, <database>ViPR</database> and </database>NCBI Database</database> Virus database.
+	For the gene expression analysis of host genes, we obtained two datasets, namely, a microarray dataset for SARS-CoV and an RNA-seq dataset for SARS-CoV-2, from the </database>NCBI Database</database> <database>GEO</database> database and <database>ViPR</database> host factor data search (Pickett et al.
+	, 2012) (<database>VIGOR4</database>) using a curated library available in the Virus Pathogen Resource(<database>ViPR</database>) (Pickett et al.
+	Databases like <database>EpiCoV</database>TM platform from  (, 2020), NCBI-SARS-CoV2 resources (NCBI-SARS-CoV2 Resources, 2020), COVID-19 data portal (<database>EMBL-EBI</database>, 2020), Virus Pathogen Database and Analysis Resource(<database>ViPR</database>) (Pickett et al.
+	Protein Identification and RetrievalThe following 12 viral protein sequences of SARS-CoV-2 were retrieved from the <database>ViPR</database> database (Host: Human, Country: India) [10]: Orf10 protein (QIA98591.
+	ResultsSelection and Retrieval of Viral Protein SequencesThe SARS-CoV-2 strain was identified and 12 viral protein sequences against the human host in India were retrieved from the <database>ViPR</database> database and selected for possible vaccine candidate identification (Table 1).
+	Table 1 SARS-CoV-2 (Host: Human, Country: India) viral protein sequence identification and retrieval via the <database>ViPR</database> database.
+	MethodsSequence alignmentNucleotide sequences of SARS-CoV-2 isolated from humans were downloaded from the National Center for Biotechnology Information (</database>NCBI Database</database>) database, the Virus Pathogen Database and Analysis Resource(<database>ViPR</database>) and from the  repository [67].
+	<database>ViPR</database> is a more professional database focusing on viruses, providing additional information, including genes, proteins, and immune epitopes, and fundamental analysis tools such as sequence alignment, phylogenetic inference, and <database>BLAST</database> comparison [30].
+	Viral data was obtained from </database>NCBI Database</database> Virus 9 with the Virus filter using each class as input, and the NIAID Virus Pathogen Database and Analysis Resource(<database>ViPR</database>) 10 through the website at http://www.
+	, 2014) and models were visualized within Jupyter notebooks with the software <database>pyViPR</database> (Ortega and Lopez, 2020).
+	, 2014) and models could be visualized within Jupyter notebooks with <database>pyViPR</database> (Ortega and Lopez, 2020).
+	, 2017)], although the user could employ atlas within a Jupyter notebook and use <database>pyViPR</database> (Ortega and Lopez, 2020) to visualize the model structure.
+	The Virus Pathogen Database and Analysis Resource(<database>ViPR</database>) (Pickett et al.
+	3 <database>Virxicon Database</database> versus other viral databasesFour existing virus databases, </database>NCBI Database</database> virus, <database>ViPR</database>, <database>ViralZone</database> and <database>Virxicon Database</database>, collect various information, and provide mechanisms that allow users to search, visualize and download it.
+	On the other hand, it does not allow to search by sequence or sequential homology—these options are implemented in </database>NCBI Database</database> and <database>ViPR</database>, both apply <database>BLAST</database> (Altschul et al.
+	</database>NCBI Database</database> virus and <database>ViPR</database> are the primary databases that allow the users to submit their data, while <database>ViralZone</database> and <database>Virxicon Database</database> make use of the data stored in the other resources.
+	The coronavirus sequence database referenced by <database>Infernal</database> was built using the <database>ViPR</database> database (44,45) (accessed in October 2020) and was composed of 25571 Coronaviridae sequences.
+	Briefly, the <database>ScanFold</database> model was used to build a structural covariation model (cm) with <database>Infernal</database> (42,43), drawing from over 25000 Coronaviridae sequences in the <database>ViPR</database> database (44,45).
+	38As an additional test of the program, 314 sequences of the spike protein for diverse β-CoV were downloaded from the <database>ViPR</database> database, and a single unaligned FASTA file used as input to the program.
+	Protein Identification and RetrievalThe following 12 viral protein sequences of SARS-CoV-2 were retrieved from the <database>ViPR</database> database (Host: Human, Country: India) [10]: Orf10 protein (QIA98591.
+	ResultsSelection and Retrieval of Viral Protein SequencesThe SARS-CoV-2 strain was identified and 12 viral protein sequences against the human host in India were retrieved from the <database>ViPR</database> database and selected for possible vaccine candidate identification (Table 1).
+	Table 1 SARS-CoV-2 (Host: Human, Country: India) viral protein sequence identification and retrieval via the <database>ViPR</database> database.
+	Viral data was obtained from </database>NCBI Database</database> Virus 9 with the Virus filter using each class as input, and the NIAID Virus Pathogen Database and Analysis Resource(<database>ViPR</database>) 10 through the website at http://www.
+	1) variants were extracted using the filters from the Virus Pathogen Resource(<database>ViPR</database>) database: severe acute respiratory syndrome coronavirus 2; human (host); complete genome only.
+	5 <database>ViPR</database> (Virus Pathogen Resource)The Virus Pathogen Resource(<database>ViPR</database>), which is operated by the National Institute of Allergy and Infectious Diseases (NIAID), is a database of pathogenic viruses.
+	Although, as of October 31st, 2022, all the data and tools from the <database>ViPR</database> database were migrated to the Bacterial and Viral Bioinformatics Resource Center (<database>BV-BRC</database> https://www.
+	1 Generalized steps of methods involved in epitope-based vaccine predictionsProtein selection and allergenicity predictionWe conducted proteomic data screening SBGNNV using <database>ViPR</database> (Pickett et al.
+	6 The miRNA prediction hits for COVID-19 proteinsInitially, the nucleotide and amino acid sequences of different SARS-CoV2 strains sourced from </database>NCBI Database</database> were queried in the Virus Pathogen Resource(<database>ViPR</database>) (https://www.
+	<database>ViPR</database> database provides comprehensive viral pathogen data (genetic and protein sequences), along with experimentally determined immune epitope data and anti-viral host responses.
+	In the next steps, <database>ViPR</database> generated SARS-CoV2 genomic data were used to predict miRNA hits (at 65% association cut-off value) against different viral proteins using <database>MirBase</database> [30].
+	MethodsSequence alignmentNucleotide sequences of SARS-CoV-2 isolated from humans were downloaded from the National Center for Biotechnology Information (</database>NCBI Database</database>) database, the Virus Pathogen Database and Analysis Resource(<database>ViPR</database>) and from the  repository [67].
+	2012) and <database>ViPR</database> databases were explored to determine essential proteins related to Orthohantavirus.
+	It was selected by analyzing <database>ViPR</database> and </database>NCBI Database</database>-<database>GenBank</database> database.
+	Authors in [17] utilized the bioinformatics tools provided in the Immune Epitope Database (<database>IEDB</database>) and Virus Pathogen Resource(<database>ViPR</database>) to identify regions corresponding to SARS-CoV-2 sequences and to predict possible epitopes.
+	<database>ViPR</database>, on the other hand, is a database containing genome, gene, and protein sequence information about human pathogenic viruses [37].
+	1 Retrieving protein sequences and identification of conserved sequences The sequences of N and G protein of different isolates of RVFV were retrieved from the <database>ViPR</database> database (Virus Pathogen Database and Analysis Resource) (http://www.
+	The Fasta formatted protein sequences were retrieved from the <database>ViPR</database> database.
+	1 Retrieval of protein sequences and identification of most conserved sequences A total of 203 sequences of the N protein (163 sequences) and G protein (40 sequences) of different isolates of RVFV were retrieved from the <database>ViPR</database> database.
+	2 Oligo design All available IBV QX-like S1 sequences (96) deposited on <database>ViPR</database> (https://www.
+	Similarly, all available Mass-type S1 sequences (166) deposited on <database>ViPR</database> (https://www.
+	They have typically made use of extensive historical experimental data about the amino acid residue sequences of epitopes such as the Epitope Database and Analysis Resource(<database>IEDB</database>) and the Virus Pathogen Resource(<database>ViPR</database>) (e.
+	Also included in these predictions are those using the Immune Epitope Database and Analysis Resource (<database>IEDB</database>) and the Virus Pathogen Resource(<database>ViPR</database>) which have already been made [54] (see later below).
+	Image 2Many of the epitopes predicted in the present study overlap with prediction made using the Immune Epitope Database and Analysis Resource (<database>IEDB</database>) and the Virus Pathogen Resource(<database>ViPR</database>) [54], and these comprised the following.
+	For phylogeny of the complete genomes of the Polish IBV strains, a preliminary analysis was carried out using all 362 gammacoronaviruses available in the NIAID Virus Pathogen Database and Analysis Resource(<database>ViPR</database>) through the website at http://www.
+	Next, 56 strains were selected for further analysis, taking into account their clustering in the <database>ViPR</database> analysis.
+	The <database>IEDB</database> and </database>NCBI Database</database>-<database>GenBank</database> database were fully deployed to analyze sequence homology, to predict targets for COV-2 in case of viral protein identification as per the related studies [14], as <database>ViPR</database> (Virus Pathogen Database Analysis Resource) are also dependent on <database>IEDB</database> and <database>GenBank</database> primarily [30].
+	We have collected and analyzed 4143 CDs of seven types of coronavirus that can infect human hosts from the virus pathogen resource (<database>ViPR</database>) (Pickett et al.
+	Similarly, using known immunogenic epitopes from the SARS-CoV-1 genome available on <database>IEDB</database> and <database>ViPR</database> immunology databases, homologous amino acids present at the equivalent position in SARS-CoV-2 have been proposed for vaccine design [44].
+	Recently, <database>ViPR</database> has developed the 2019 Novel Coronavirus (2019-nCoV) portal in <database>ViPR</database> to support research efforts on this emerging public health threat (https://www.
+	3 Virus Pathogen Resource(<database>ViPR</database>) The Virus Pathogen Resource(<database>ViPR</database>; http://www.
+	Currently, <database>ViPR</database> contains major pathogenic viruses with single-stranded positive-sense RNA belonging to the family (Togaviridae, Caliciviridae, Coronaviridae, Hepeviridae, Flaviviridae, and Picornaviridae); single-stranded negative-sense RNA (Paramyxoviridae, Bunyavirales, Filoviridae, Pneumoviridae, and Rhabdoviridae); double-stranded RNA (Reoviridae); double-stranded DNA (Herpesviridae and Poxviridae).
+	The ‘Workbench’ feature of <database>ViPR</database> allows users to store and share results for future use.
+	spg+decorator=influenza, ) [69] like the <database>ViPR</database>, is an NIAID funded resources that support the research of viral pathogens causing (re)emerging infectious diseases.
+	 and <database>ViPR</database> integrate data from external sources (like <database>GenBank</database>, <database>UniProt</database>, <database>IEDB</database>, protein data Bank, etc.
+	Thedatabase along with the <database>ViPR</database> maintains and updates the novel coronavirus portal hosted in <database>ViPR</database> (https://www.
+	<database>ViPR</database> was the available resource to provide the bioinformatics resource for applying to generate the data in this study.
+	After alignment, the tree was generated for entire isolates based nucleotide variations with the NCBI-SARS-CoV-2 resources and Virus Pathogen Database and Analysis Resource(<database>ViPR</database>) [36]).
+	3 Analysis of genomic variationsWe computed the extent of variation at each position of nucleotide sequences uploaded to the <database>ViPR</database> genome variation analysis tool and visualized aligned nucleotide sequences using the <database>ViPR</database> interactive alignment viewer.
+	Aligned sequences generated from the <database>ViPR</database> tool were further viewed in the <database>BioEDIT</database> v7.
+	On the other hand, web applications are often designed to submit jobs on behalf of a set of registered users (eg, <database>ViPR</database>,9 see below), but it is disruptive to ask each user to provide additional (CRA) authentication for each submission.
+	Usage Examples Three web applications currently are using the CRA to provide access to <database>CIPRES</database> resources: <database>MorphoBank</database>,10,11 ,8 and <database>ViPR</database>.
+	Theand <database>ViPR</database> are public database and analysis resources for the study of influenza viruses and Category A–C priority pathogens, respectively.
+	Theand <database>ViPR</database> web applications natively support tree inference and other computational biology methods using several community tools.
+	 and <database>ViPR</database> have historically provided web forms and required infrastructure to manage job submissions and results retrieval.
+	However, with the continued growth of available data, /<database>ViPR</database> developers sought a mechanism to support analysis tasks that exceeded the capacity of their available compute resources.
+	Using the CRA, /<database>ViPR</database> now facilitates direct access to the <database>RAxML</database> tool on <database>XSEDE</database> (<database>RAXMLHPC2_TGB</database>) through their existing phylogenetic tree analysis portal.
+	The tree analysis input page in /<database>ViPR</database> allows users to configure parameters for <database>RAxML</database> runs.
+	On submission, when the number and size of the input sequences exceeds a prespecified threshold, a light-box appears to recommend that the user submit their job to <database>CIPRES</database> through the existing /<database>ViPR</database> interface (Fig.
+	A <database>REST</database> client using the <database>Jersey</database> framework implemented in /<database>ViPR</database> then communicates with the RESTful services at <database>CIPRES</database>.
+	After jobs are successfully submitted to <database>CIPRES</database>,and <database>ViPR</database> periodically poll the job status and retrieve the results once the jobs are finished.
+	The results are then saved to the user’s private workbench area inor <database>ViPR</database> for further use.
+	Thus, the results obtained either from localand <database>VIPR</database> or <database>CIPRES</database> resources can be accessed, visualized, and analyzed seamlessly by the user without ever leaving theand <database>ViPR</database> web sites.
+	Figure 2 Screen shot showing /<database>ViPR</database> lightbox pop up, prompting user to choose the option of running the analysis task using the <database>CIPRES</database> resource.